src/dynamo0p3.py

```diff
@@ -2949,7 +2960,18 @@ def generate(self):
         # Add boundary dofs array to the operator boundary condition
         # kernel (enforce_operator_bc_kernel) arguments
         if self._kern.name.lower() == "enforce_operator_bc_code":
-            self.operator_bcs_kernel(fs_to)
+            # Minimal sanity check - this kernel should only have a single LMA
+            # operator as argument so the first one must be a gh_operator.
+            # Full sanity checking is done in the operator_bcs_kernel()
```

Comment:
I've just remembered why I did it this way - I was concerned that the implementation of each method should stand on its own two feet and validate its arguments. If we pull the validation checks out and put them in the base class that calls them then, should we ever add additional calls to them from somewhere else, we will lose the validation checks. On the other hand I suspect we are unlikely to call them from elsewhere and we can add comments documenting the checks. A third possibility is to put some sort of `validate` method in the base class and call that to do the checks.